quic/s2n-quic-qns/src/client/interop.rs

```diff
@@ -83,13 +83,21 @@ impl Interop {
 
         let endpoints = self.endpoints().await?;
 
-        let mut tasks = task::Limiter::new(self.concurrency);
+        // In the Resumption test, the client needs to start a handshake and complete it before
+        // starting a resumption handshake with a session ticket from the first handshake. Therefore,
+        // the two handshake requests are run one after another synchronously.
+        let concurrency = if matches!(self.testcase, Some(Testcase::Resumption)) {
+            1
+        } else {
+            self.concurrency
+        };
+        let mut tasks = task::Limiter::new(concurrency);
 
         // https://github.com/marten-seemann/quic-interop-runner#test-cases
         // Handshake Loss (multiconnect): Tests resilience of the handshake to high loss.
         // The client is expected to establish multiple connections, sequential or in parallel,
         // and use each connection to download a single file.
-        if let Some(Testcase::Multiconnect) = self.testcase {
+        if let Some(Testcase::Multiconnect) | Some(Testcase::Resumption) = self.testcase {
```

Comment:
very minor nit: you should be able to combine the match
```suggestion
        if let Some(Testcase::Multiconnect | Testcase::Resumption) = self.testcase {
```